TESTed (#124)
dodona-edu/dodona-edu.github.io: nl/guides/teachers/tested-template-exercise/README.md

---

> title: "Sjabloonoefening TESTed"

Reviewer: ```suggestion
title: "Oefeningssjabloon TESTed"
```

> description: "Tutorial: Sjabloonoefening TESTed"

Reviewer: ```suggestion
description: "Tutorial: Oefeningssjabloon TESTed"
```

> ---

Reviewer: Basically replace "sjabloonoefening" bij "oefeningssjabloon" everywhere.

::: warning Opmerking
Sjabloonoefeningen worden gebruikt wanneer je één oefening in meerdere programmeertalen wilt aanbieden.
:::

# Sjabloonoefening TESTed

> De TESTed judge is een programmeertaal onafhankelijke judge,
> dit betekent dat TESTed meerdere programmeertalen ondersteund.
> Dit is mogelijk door het opstellen van programmeertaal onafhankelijke oefeningen: je schrijf één oefening,
> welke oplosbaar is in meerdere programmeertalen (zie de [TESTed-referentie](../../../references/tested-judge/)).

Reviewer: ```suggestion
TESTed is een meertalige judge, waarmee we bedoelen dat TESTed meerdere programmeertalen ondersteunt.
Dit maakt het mogelijk om programmeertaalonafhankelijke oefeningen op te stellen: je schrijf één oefening,
die oplosbaar zal zijn in meerdere programmeertalen (zie de [TESTed-referentie](../../../references/tested-judge/)).
```

> Hoewel de judge programmeertaal onafhankelijk is, verwacht Dodona voorlopig per programmeertaal een aparte oefening.

Reviewer: ```suggestion
Hoewel de judge meerdere programmeertalen ondersteunt, verwacht Dodona voorlopig per programmeertaal een aparte oefening.
```

Als oplossing hiervoor introduceert deze handleiding het concept van sjabloonoefeningen.

> Dit is een programmeertaal onafhankelijke oefening beschrijving,

Reviewer: ```suggestion
Dit is een programmeertaalonafhankelijk sjabloon voor een oefening,
```

> welke zal worden omgezet naar een programmeertaal specifieke (en Dodona compatibele) oefening voor TESTed.

Reviewer: ```suggestion
dat zal worden omgezet naar een programmeertaalspecifieke (en Dodona-compatibele) oefening voor TESTed.
```

## 1. Aanmaken Git repository
De oefeningen voor Dodona bevinden zich in een Git repository, zoals normale oefeningen.
We verwijzen naar [de handleiding _oefening repositories aanmaken_](../new-exercise-repo).

## 2. Mapstructuur
De mapstructuur voor een sjabloonoefening van TESTed komt in grote mate overeen met de
[vereiste structuur door Dodona](../../../references/exercise-directory-structure).
Bij de sjabloonoefening is het `config.json` bestand hernoemd naar `config.template.json`,
met als hoofdreden dat we de sjabloonoefening niet als een oefening op Dodona willen weergeven.

### Voorbeeld minimale mapstructuur
```text
+-- sjabloon/oefening/map
|   +-- config.template.json       # Configuratie sjabloon oefening
|   +-- evaluation                 #
|   |   +-- plan.yaml              # DSL-testplan
|   +-- description                #
|   |   +-- description.nl.md.mako # Markdown sjabloonopgave nederlands
|   |   +-- description.en.md.mako # Markdown sjabloonopgave engels
:   :   :
```

## 3. Opstellen testplan

> De testen voor in oefening in TESTed worden geschreven in een testplan.

Reviewer: ```suggestion
De testen voor een oefening in TESTed worden geschreven in een testplan.
```

Documentatie over hoe een testplan kan worden opgesteld,
kan gevonden worden op [TESTed DSL-testplannen](../../../references/tested-judge/dsl).
We veronderstellen dat dit testplan zich bevindt in het bestand `evaluation/plan.yaml`.

::: tip Tip voor geavanceerde gebruikers
De sjabloonoefeningen kunnen ook gebruikmaken van de
[geavanceerde testplannen](../../../references/tested-judge/json).
:::

## 4. Opstellen sjabloonopgaven

> De sjabloonopgaven worden geschreven met het Mako sjablonensysteem,

Reviewer: ```suggestion
De sjabloonopgaven worden geschreven met het sjablonensysteem Mako,
```
